pilot/pkg/config/memory/controller.go

```diff
@@ -26,9 +26,35 @@ import (
 	"istio.io/istio/pkg/config/schema/collection"
 )
 
+type Config struct {
+	HasSynced func() bool
+	SyncProc  bool
```

Comment:
what's this?

Reply:
It keeps the ability to new a controller with sync monitor. U can see the code below.

Reply:
Yes, understood. Looks below is more readable

func NewController(cs model.ConfigStore, hasSynced func() bool) model.ConfigStoreCache {

Reply:
I think the `hasSynced` func is not a common args. It's more like a customization. So I didn't add it to the **common** `NewController` func.
BTW, there've been tons of usage of `NewController`.